core/config/toml/types.go

```diff
@@ -1301,3 +1303,61 @@ func (t *ThresholdKeyShareSecrets) validateMerge(f *ThresholdKeyShareSecrets) (e
 
 	return err
 }
+
+type Tracing struct {
+	Enabled         *bool
+	CollectorTarget *string
+	NodeID          *string
+	SamplingRatio   *float64
+	Attributes      *map[string]string
+}
+
+func (t *Tracing) setFrom(f *Tracing) {
+	if v := f.Enabled; v != nil {
+		t.Enabled = f.Enabled
+	}
+	if v := f.CollectorTarget; v != nil {
+		t.CollectorTarget = f.CollectorTarget
+	}
+	if v := f.NodeID; v != nil {
+		t.NodeID = f.NodeID
+	}
+	if v := f.Attributes; v != nil {
+		t.Attributes = f.Attributes
+	}
+	if v := f.SamplingRatio; v != nil {
+		t.SamplingRatio = f.SamplingRatio
+	}
+}
+
+func (t *Tracing) ValidateConfig() (err error) {
+	if t.CollectorTarget == nil || *t.CollectorTarget == "" {
+		return nil
```

Comment:
Should we reject empty string? That is an explicit misconfiguration, isn't it?

Reply:
In the last revision I included the check for unit testing - empty string is the zero value of string parameters, which is needed for our tstar tests in the root directory

Reply:
Sorry I'm not following. nil is the zero value for string pointers. empty string should only appear here if we are parsing an explicit `CollectorTarget = ""`. What do tstar tests have to do with anything? We shouldn't have to compromise the UX for test quirks.

Reply:
I'm not quite sure why the empty string is showing up when the `CollectorTarget` is not specified yet, but that's the behavior i'm observing for our tstar tests.